SalesforceFoundation/EDA
# [EDA]: Prevent Deletion of Course Connection if There are Associated Child Records

# Critical Changes

# Changes
## Updates to Ensure Data Integrity
We've added functionality to prevent the deletion of Course Connection records that have related Term Grade or Attendance Event records. This change is for internal-use only and is disabled at this time. You won't see any difference in functionality.
**Note:** We strongly recommend that EDA Admins don't manage Hierarchy Custom Settings directly and, instead, use the EDA Settings UI.

# Issues Closed
Closes #1046
# New Metadata
- CCON_CannotDelete_TDTM
- CCON_CannotDelete_TEST

# Deleted Metadata

# Testing Notes

* Scenario 1: “Prevent Course Connection Deletion” checkbox is enabled and Course Connection record has related Term Grade or Attendance Event records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create a Contact with a Course Connection. 
    * Add a Term Grade or Attendance Event or both to the Course Connection record. 
    * Delete the Course Connection record. 
    * You should receive a pop up error that you cannot delete this Course Connection. See https://quip.com/FUGPAUO0JSbY#UBKACA9I3wO for the error message. 

* Scenario 2: “Prevent Course Connection Deletion” checkbox is enabled and Course Connection record does not have related Term Grade or Attendance Event records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create a Contact with a Course Connection. 
    * Delete the Course Connection record. 
    * You should be able to delete Course Connection record. 

* Scenario 3: “Prevent Course Connection Deletion” checkbox is disabled and Course Connection record has related Term Grade or Attendance Event records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create a Contact with a Course Connection. 
    * Add a Term Grade or Attendance Event or both to the Course Connection record. 
    * Delete the Course Connection record. 
    * You should be able to delete Course Connection record. 

* Scenario 4: “Prevent Course Connection Deletion” checkbox is disabled and Course Connection record does not have related Term Grade or Attendance Event records. 
    * Check “Prevent Course Connection Deletion” checkbox in Hierarchy Settings. 
    * Create a Contact with a Course Connection. 
    * Delete the Course Connection record. 
    * You should be able to delete Course Connection record.
